src/directory/file_slice.rs

```diff
@@ -2,10 +2,11 @@ use stable_deref_trait::StableDeref;
 
 use crate::common::HasLen;
 use crate::directory::OwnedBytes;
-use std::sync::Arc;
+use std::sync::{Arc, Weak};
 use std::{io, ops::Deref};
 
-pub type BoxedData = Box<dyn Deref<Target = [u8]> + Send + Sync + 'static>;
+pub type BoxedData = Arc<dyn Deref<Target = [u8]> + Send + Sync + 'static>;
```

Comment:
Let me know if you want to change the names of `BoxedData` and `WeakBoxedData` for something else.

Reply:
`ArcBytes` and `WeakArcBytes` maybe?